src/overlays/actors/ovl_En_Ge2/z_en_ge2.c

```diff
@@ -39,78 +35,738 @@ const ActorInit En_Ge2_InitVars = {
+                    Animation_Change(&this->skelAnime, &gGerudoPurpleRunningAwayCutsceneAnim, 1.0f, 0.0f,
+                                     Animation_GetLastFrame(&gGerudoPurpleRunningAwayCutsceneAnim), 0, -5.0f);
+                    this->screamTimer = (s32)(Rand_ZeroFloat(10.0f) + 20.0f);
+                    break;
+
+                case ENGE2_CSACTION_BEEHIVE_EXIT:
+                    Actor_MarkForDeath(&this->picto.actor);
+                    break;
+
+                case ENGE2_CSACTION_GBT_ENTR_STAND_STILL:
+                    Animation_Change(&this->skelAnime, &gGerudoPurpleGreatBayCutsceneAnim, 0.0f, 0.0f, 0.0f, 2, 0.0f);
+                    break;
+
+                case ENGE2_CSACTION_GBT_ENTR_BLOWN_AWAY:
+                    Animation_Change(&this->skelAnime, &gGerudoPurpleGreatBayCutsceneAnim, 0.0f, 1.0f, 1.0f, 2, 0.0f);
+                    EnGe2_SetupBlownAwayPath(this, play);
+                    this->stateFlags |= GERUDO_PURPLE_STATE_DISABLE_MOVEMENT;
+                    this->screamTimer = (s32)(Rand_ZeroFloat(10.0f) + 20.0f);
+                    break;
+
+                default:
+                    break;
+            }
+        }
+    }
+
+    switch (this->csAction) {
+        case ENGE2_CSACTION_BEEHIVE_RUN_AWAY:
+            EnGe2_FollowPath(this);
+            this->picto.actor.speedXZ = 5.0f;
+
+            if (Animation_OnFrame(&this->skelAnime, 2.0f) || Animation_OnFrame(&this->skelAnime, 6.0f)) {
+                Actor_PlaySfxAtPos(&this->picto.actor, NA_SE_EV_PIRATE_WALK);
+            }
+
+            if (this->screamTimer > 0) {
+                this->screamTimer--;
+            } else {
+                this->screamTimer = (s32)(Rand_ZeroFloat(10.0f) + 20.0f);
+                EnGe2_Scream(this);
+            }
+            break;
+
+        case ENGE2_CSACTION_GBT_ENTR_BLOWN_AWAY:
+            if ((this->curPointIndex < this->path->count) && EnGe2_FollowPathWithoutGravity(this)) {
+                this->curPointIndex++;
+            }
+
+            // Tumble in the air
+            this->picto.actor.shape.rot.x += 0x3E8;
+            this->picto.actor.shape.rot.y += 0x7D0;
+            this->picto.actor.shape.rot.z += 0x1F4;
+
+            if (this->screamTimer > 0) {
+                this->screamTimer--;
+            } else {
+                this->screamTimer = (s32)(Rand_ZeroFloat(10.0f) + 20.0f);
+                EnGe2_Scream(this);
+            }
+            break;
+
+        default:
+            break;
+    }
+}
+
+// Used for those on boats
+void EnGe2_GuardStationary(EnGe2* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+    if (EnGe2_LookForPlayer(play, &this->picto.actor, &this->picto.actor.focus.pos, this->picto.actor.shape.rot.y,
+                            0x4000, 720.0f, this->verticalDetectRange)) {
+        if ((GERUDO_PURPLE_GET_EXIT(&this->picto.actor) != GERUDO_PURPLE_EXIT_NONE) && !Play_InCsMode(play)) {
```

Comment:
It seems so weird that they even bother to check for `if exit == none` without code to handle that case, they just... stare and turn aggressively at you, no talking or anything, kinda weird.